include/oneapi/dpl/internal/serial_numeric_impl.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) Intel Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef _ONEDPL_SERIAL_NUMERIC_IMPL_H
+#define _ONEDPL_SERIAL_NUMERIC_IMPL_H
+#include <iterator>
+
+namespace oneapi
+{
+namespace dpl
+{
+//namespace serial {
+
+template <class _InputIterator, class _Tp, class _BinaryOp>
+_Tp
+reduce(_InputIterator __first, _InputIterator __last, _Tp __init, _BinaryOp __b)
```

Comment:
These algorithms are available since C++17. I believe we should have a check
```cpp
#if  __cplusplus < 201703
// implement algorithms
#endif
```

Reply:
Hi, @capatober 
If so, when developers use -std=c++20 or newer, serial reduce won't be available in oneapi::dpl.

Reply:
It could be 
```cpp
#if  __cplusplus < 201703
// implement algorithms
#else
namespace oneapi
{
namespace dpl
{
using ::std::reduce;
} // namespace dpl
} // namespace oneapi
#endif
```

Reply:
@capatober 
I find "using ::std::reduce;" here may lead to compfail, not sure whether it is a oneDPL issue, if we use following code in oneapi/dpl/numeric:
```c++
....
namespace std
{
...
using oneapi::dpl::reduce;
...
}

namespace oneapi
{
namespace dpl
{
using ::std::reduce;
}
}
```
Following reproducer will compfail:
reproducer.cpp
#include <oneapi/dpl/numeric>
#include <oneapi/dpl/execution>

clang++ -fsycl -c reproduer.cpp
Thank very much.

Reply:
Do you mean this happens with this patch or with the `main` branch?

Reply:
I mean main branch with the change injecting std reduce to oneapi dpl namespace.
Thanks very much.